src/openms/include/OpenMS/ANALYSIS/ID/IDScoreSwitcherAlgorithm.h

```diff
@@ -552,6 +552,145 @@ namespace OpenMS
       }
     }
 
+
+  struct IDSwitchResult
+  {
+    // the score name, orientation and type used before the switch
+    String original_score_name;
+    bool original_score_higher_better = true;
+    IDScoreSwitcherAlgorithm::ScoreType original_score_type = IDScoreSwitcherAlgorithm::ScoreType::RAW;
+    // the score name, orientation and type used after the switch
+    bool requested_score_higher_better = original_score_higher_better;    
+    IDScoreSwitcherAlgorithm::ScoreType requested_score_type = original_score_type;      
+    String requested_score_name; // the search engine score name (e.g. "X!Tandem_score" or score category (e.g. "PEP")
+    // wheter the main score was switched
+    bool score_switched = false;
+  };
+
+  static IDSwitchResult switchToScoreType(ConsensusMap& cmap, String requested_score_type_as_string, bool include_unassigned = true)
```

Comment:
moved to IDScoreSwitcher for general use.
now has:
switchToScoreType
switchBackScoreType

to set and reset a score type